src/osx/core/strconv_cf.cpp

```diff
@@ -89,27 +89,23 @@ WXDLLIMPEXP_BASE wxMBConv* new_wxMBConv_cf(wxFontEncoding encoding)
         if ( theString == NULL )
             return wxCONV_FAILED;
 
+        if ( m_compose_decompose )
```

Comment:
I don't really understand why we don't do this for "plain" `wxMBConv_cf` any more... I think the comment below is still relevant, i.e. we do want to use NFC internally -- but now we don't, unless we're working with file paths.

Reply:
Wasn't the reason for normalizing the input from filenames, so this would be wxConvFilename ? I thought that having round-trip fidelity in the normal cf conversion would be good. But I can leave it on of course

Reply:
I'm not sure why was this done initially, but, as I wrote on the list, `wxMBConv_cf` is used by `wxCSConv`, i.e. this is going to change the behaviour of any code using this (public) class and it doesn't look like a good idea, especially because the old behaviour (outputting NFC) seems more useful to me.

For round-trip fidelity, we could (should?) consider using NFC in `ToWChar()` too, unless the new flag (bool or enum) indicates that NFD should be used.

Reply:
I was thinking about roundtrip fidelity (not for the filename converter, but the 'normal' one) in the sense that a decomposed utf8 would be still decomposed in the wxString, and then when written back it would still be decomposed, exactly as a composed utf8 would be composed wxString and written out composed. So maintaining the normalization throughout the conversion.

APFS once had a version that that only opened the files if the normalization was correct, so as with case sensitive filesystem, it was normalization sensitive. You could have to identical looking filenames one composed one decomposed, eg one typed in the finder decomposed, another one copied from a windows file system composed. You had to preserve normalization otherwise you would open the wrong file. But this situation is better now, it is now normalization-insensitive ...

Reply:
Note that `wxString` never stores UTF-8 in the default build (I agree that what you say would be a concern for `wxUSE_UNICODE_UTF8` build, but AFAIK nobody uses it anyhow, so...). This is why I say that NFC only/mostly matters in `ToWChar()`.